File: backend/src/main/java/com/auto_mendes/backend/entity/Vehicle.java
package com.auto_mendes.backend.entity;

import java.io.Serializable;
import java.math.BigDecimal;
import java.util.List;

import com.auto_mendes.backend.enums.TransmissionType;

import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Vehicle implements Serializable {
	private static final long serialVersionUID = 1L;
	
	private String id;
	private String plate;
	@Enumerated(EnumType.STRING)
	private TransmissionType transmissionType;
	private BigDecimal price;
	private Model model;
	private List<SaleVehicle> saleVehicles;
}

File: frontend/src/app/model/vehicle.ts
import { TransmissionType } from "../enum/transmission_type";
import { VehicleType } from "../enum/vehicle_type";

export interface Vehicle {
    id: string,
    plate: string,
    price: number,
    model: Model,
    transmissionType: TransmissionType,
    vehicleType: VehicleType,
    saleVehicles: Sale[] 
} 